src/api/v2/v2AzureResourcesApi.ts

```diff
@@ -73,6 +73,8 @@ export interface ResourceQuickPickOptions {
 }
 
 export interface ResourceModelBase {
+    // TODO: Should IDs be optional?
```

Comment:
I'd love if we could keep this optional. Label is a required result from `getTreeItem`, perhaps we can use that if nothing else is specified? VSCode uses the label as (at least part of) the ID if it isn't specified, though I've not seen the actual code there.

Reply:
A bonus to keeping this and the other values optional is that the model objects don't necessarily have to be `object`s. They could be e.g. `string`. I've really hoped we could keep that true.

(As a related aside, I think when applicable, e.g. when we're checking if the model object has an `azureResourceId` or `azurePortalUrl` or whatever, we should first check if the model object is actually an `object`--make no assumptions about what those models are)

Reply:
I can't think of a reason why ID *must* be specified; i.e. why we can't just choose an arbitrary unique value if omitted.  It's obviously *better* if BDPs use proper IDs, but there are definately cases where it's just unlikely to matter (e.g. generic nodes, error nodes, etc.).

We can't use the tree item ID--at least, not generally--as we don't necessarily have the tree item, as we're working solely with model objects.  It would probably make sense, if BDPs give their tree items IDs, that they use the same ID as the model itself.  If we let VS Code be the sole interpretor of tree item IDs, then we could leave that as a "best practice" but not necessarily care.